pallets/creator-staking/src/lib.rs

```diff
@@ -0,0 +1,748 @@
+
+            let staking_info = Self::creator_stake_info(space_id, era).unwrap_or_default();
+            let reward_and_stake =
+                Self::general_era_info(era).ok_or(Error::<T>::UnknownEraReward)?;
+
+            let (_, stakers_combined_reward_share) =
+                Self::distributed_rewards_between_creator_and_stakers(&staking_info, &reward_and_stake);
+            let staker_reward =
+                Perbill::from_rational(staked, staking_info.total) * stakers_combined_reward_share;
+
+            let mut ledger = Self::ledger(&staker);
+
+            let should_restake_reward = Self::should_restake_reward(
+                restake,
+                creator_info.state,
+                staker_info.latest_staked_value(),
+            );
+
+            if should_restake_reward {
+                staker_info
+                    .stake(current_era, staker_reward)
+                    .map_err(|_| Error::<T>::UnexpectedStakeInfoEra)?;
+
+                // Restaking will, in the worst case, remove one, and add one record,
+                // so it's fine if the vector is full
+                ensure!(
+                    staker_info.len() <= T::MaxEraStakeValues::get(),
+                    Error::<T>::TooManyEraStakeValues
+                );
+            }
+
+            // Withdraw reward funds from rewards holding account
+            let reward_imbalance = T::Currency::withdraw(
+                &Self::account_id(),
+                staker_reward,
+                WithdrawReasons::TRANSFER,
+                ExistenceRequirement::AllowDeath,
+            )?;
+
+            if should_restake_reward {
+                ledger.locked = ledger.locked.saturating_add(staker_reward);
+                Self::update_ledger(&staker, ledger);
+
+                // Update storage
+                GeneralEraInfo::<T>::mutate(current_era, |value| {
+                    if let Some(x) = value {
+                        x.staked = x.staked.saturating_add(staker_reward);
+                        x.locked = x.locked.saturating_add(staker_reward);
+                    }
+                });
+
+                CreatorEraStake::<T>::mutate(space_id, current_era, |staking_info| {
+                    if let Some(x) = staking_info {
+                        x.total = x.total.saturating_add(staker_reward);
+                    }
+                });
+
+                Self::deposit_event(Event::<T>::Staked {
+                    who: staker.clone(),
+                    creator: space_id,
+                    era: current_era,
+                    amount: staker_reward,
+                });
+            }
+
+            // TODO: mint tokens to balance before locking in case `should_restake_reward` is true
+            T::Currency::resolve_creating(&staker, reward_imbalance);
+            Self::update_staker_info(&staker, space_id, staker_info);
+
+            // TODO: change event
+            Self::deposit_event(Event::<T>::RewardsClaimed {
```

Comment:
Need to be more specific:
```suggestion
            Self::deposit_event(Event::<T>::StakerRewardsClaimed {
```